python/src/acp_sdk/client/client.py

```diff
@@ -170,7 +173,12 @@ def _raise_error(self, response: httpx.Response) -> None:
         try:
             response.raise_for_status()
         except httpx.HTTPError:
-            raise ACPError(Error.model_validate(response.json()))
+            data = response.json()
+            try:
+                error = Error.model_validate(data)
+            except ValidationError:
+                error = Error(code=ErrorCode.SERVER_ERROR, message=str(data))
```

Comment:
I'd rather leave this without translation. It is a either a programming error on the server, or some version incompatibility. Either way masking it as usual server error will be harder to diagnose in my opinion.

Reply:
you will get `ValidationError` with no information from the line above instead, thats worse imo :D

Reply:
should I raise it as generic `RuntimeError(str(data))` instead?

Reply:
there might be, not sure if you'll get the contents always
but you're right, these non-conforming errors would be bugs we should fix, not a normal situation, reverting back